include/libsemigroups/transf.hpp

```diff
@@ -49,705 +55,642 @@
+    //!
+    //! Constructs a partial transformation initialized using the
+    //! container \p cont as follows: the image of the point \c i under
+    //! the partial transformation is the value in position \c i of the
+    //! container \p cont.
+    //!
+    //! \param cont the container.
+    //!
+    //! \exceptions
+    //! \no_libsemigroups_except_detail
+    //!
+    //! \complexity
+    //! Linear in the size of the container \p cont.
+    //!
+    //! \warning
+    //! No checks on the validity of \p cont are performed.
+    explicit PTransfBase(Container const& cont) : _container(cont) {}
 
-      //! Compare for greater than or equal.
-      //!
-      //! Returns \c true if `*this` is greater than or equal to \p that by
-      //! comparing the image values of `*this` and \p that.
-      //!
-      //! \param that the partial transformation for comparison.
-      //!
-      //! \returns
-      //! A value of type \c bool.
-      //!
-      //! \exceptions
-      //! \no_libsemigroups_except_detail
-      //!
-      //! \complexity
-      //! At worst linear in degree().
-      bool operator>=(PTransfBase const& that) const {
-        return that <= *this;
-      }
+    //! Construct from an initializer list.
+    //!
+    //! Constructs a partial transformation initialized using the
+    //! container \p cont as follows: the image of the point \c i under
+    //! the partial transformation is the value in position \c i of the
+    //! container \p cont. The values in the initializer list must be
+    //! convertible to point_type or equal to \ref UNDEFINED.
+    //!
+    //! \param cont the initializer list.
+    //!
+    //! \exceptions
+    //! \no_libsemigroups_except_detail
+    //!
+    //! \complexity
+    //! Linear in the size of the initializer list \p cont.
+    //!
+    //! \warning
+    //! No checks on the validity of \p cont are performed.
+    //!
+    //! \sa
+    //! \ref make
+    template <typename T>
+    explicit PTransfBase(std::initializer_list<T> cont) : PTransfBase() {
+      static_assert(std::is_same_v<T, Undefined>
+                        || std::is_convertible_v<T, point_type>,
+                    "the template parameter T must be Undefined or "
+                    "convertible to point_type!");
+      resize(_container, cont.size());
+      std::copy(cont.begin(), cont.end(), _container.begin());
+    }
 
-      //! Compare for inequality.
-      //!
-      //! Returns \c true if `*this` does not equal \p that by comparing the
-      //! image values of `*this` and \p that.
-      //!
-      //! \param that the partial transformation for comparison.
-      //!
-      //! \returns
-      //! A value of type \c bool.
-      //!
-      //! \exceptions
-      //! \no_libsemigroups_except_detail
-      //!
-      //! \complexity
-      //! At worst linear in degree().
-      bool operator!=(PTransfBase const& that) const {
-        return !(*this == that);
-      }
+    // TODO doc
```

Comment:
Should this be documented, or is this being left for later?

Reply:
Should be documented, I'll add that, thanks!